### PR TITLE
Named axis support

### PR DESCRIPTION
**Pull request by thierryH91200**
capture d ecran 2017-04-24 a 20 40 57

capture d ecran 2017-04-24 a 20 45 41

capture d ecran 2017-04-24 a 20 52 39

add
 ```
      axis.nameAxisFont = NSUIFont.systemFont(ofSize: CGFloat(14.0))
        axis.nameAxis = "Day"
        axis.nameAxisTextColor = #colorLiteral(red: 1, green: 0.1491314173, blue: 0, alpha: 1)
        axis.nameAxisEnabled = true

```

I think it should have success like PR

### CODE REVIEW

**Comment by liuxuan30** [comment]
Wow, you have been helping a lot!
first question, do you consider adding some offset for the left/right of the name? (bottom/top for x axis).

**Comment by liuxuan30** [comment]
Wow, you have been helping a lot!
first question, do you consider adding some offset for the left/right of the name? (bottom/top for x axis).

Supporting only 90 degree at beginning is fine.

File: Source/Charts/Renderers/YAxisRenderer.swift [hunk 1]
```diff
@@ -53,20 +54,25 @@ open class YAxisRenderer: AxisRendererBase
...
-            
         }
         else
         {
             if labelPosition == .outsideChart
             {
                 textAlign = .left
                 xPos = viewPortHandler.contentRight + xoffset
+                
+                xPosName = viewPortHandler.chartWidth - 5
```
Comment (liuxuan30) [inline Source/Charts/Renderers/YAxisRenderer.swift]: what is 5 here?

File: Source/Charts/Renderers/YAxisRenderer.swift [hunk 2]
```diff
@@ -44,6 +44,7 @@ open class YAxisRenderer: AxisRendererBase
         let labelPosition = yAxis.labelPosition
         
         var xPos = CGFloat(0.0)
+        var xPosName = CGFloat(0.0)
```
Comment (liuxuan30) [inline Source/Charts/Renderers/YAxisRenderer.swift]: `xPosName` is a little confusing here. how about `xNamePos` or other style?

File: Source/Charts/Charts/BarLineChartViewBase.swift [hunk 3]
```diff
@@ -426,38 +426,68 @@ open class BarLineChartViewBase: ChartViewBase, BarLineScatterCandleBubbleChartD
...
+                    
+                }
             }
-
+            
             if xAxis.isEnabled && xAxis.isDrawLabelsEnabled
             {
                 let xlabelheight = xAxis.labelRotatedHeight + xAxis.yOffset
                 
+                var size = NSSize()
```
Comment (liuxuan30) [inline Source/Charts/Charts/BarLineChartViewBase.swift]: Travis CI for tvOS and iOS fails here, maybe NSUISize()?

**Comment by thierryH91200** [comment]
1 do you consider adding some offset for the left/right of the name? (bottom/top for x axis).
**yes i add offset depending on the height of the font**

capture d ecran 2017-04-25 a 08 00 24

2 seems you forgot horizontal bar chart and radar/pie? Seeing separate axis renderer for them. Maybe we can consolidate in a common place so we can reuse them
**I'll take care of it**

**Comment by liuxuan30** [comment]
Oh I forgot. Seems we can ignore pie/radar chart's for now, since their axis is not normal, and we don't have requests for displaying name?

**Comment by liuxuan30** [comment]
@thierryH91200 I was thinking, if there are demands, such as the inter space between the name and axis line? I see you are just using existing `offsetLeft/right/top/bottom` for the space between the chart view boundary and the axis name.

I am not sure if we should use separate "axisName Left/Right/Top/Bottom/ Space" for them. Any other people who have ideas?

**Comment by thierryH91200** [comment]
add HorizontalBarChart

capture d ecran 2017-04-26 a 15 09 50

**Comment by thierryH91200** [comment]
I was not very pleased with me
Complete rewrite of name Axis
Creation of 4 rectangles to preserve the place of the name of the axis
This avoids overlays and has a more readable code

     Open var nameRectBottom = CGRect ()
     Open var nameRectTop = CGRect ()
     Open var nameRectLeft = CGRect ()
     Open var nameRectRight = CGRect ()

For BarLineChartViewBase
NameRectBottom and nameRectTop are associated with xAxis
NameRectLeft is associated with leftAxis
NameRectRight are associated with rightAxis

For HorizontalBarChartView
NameRectLeft and nameRectRight are associated with xAxis
NameRectBottom is associated with rightAxis
NameRectTop is associated with leftAxis

**Comment by liuxuan30** [comment]
LOL, take your time:)

**Comment by hungcaobss** [comment]
Hi, was this supported for iOS? I have checked in the latest version but don't see it

**Comment by thierryH91200** [comment]
not reviewed yet.

capture d ecran 2017-05-22 a 08 06 04

**Comment by seenoevo** [comment]
please add this feature @danielgindi :)

**Comment by dhaneshgosai** [comment]
Hello  @danielgindi,

Is this feature reviewed ?

**Comment by shashidhar34** [comment]
is this feature in the current released version?

**Pull request closed without merging**